tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,331 @@
+		},
+	}
+	return backup
+}
+
+func setRestoreObject() bv1.Restore {
+	broConfig := backupRestoreConfig
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, broConfig)
+
+	restore := bv1.Restore{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "restore-",
+		},
+		Spec: bv1.RestoreSpec{
+			BackupFilename: backupName,
+			Prune:          &broConfig.Prune,
+		},
+	}
+	return restore
+}
+
+func installBroChart(client *rancher.Client) error {
+	project, err := projects.GetProjectByName(client, cluster, "System")
+	if err != nil {
+		return err
+	}
+
+	cluster, err := shepClusters.NewClusterMeta(client, cluster)
+	if err != nil {
+		return err
+	}
+
+	latestBackupVersion, err := client.Catalog.GetLatestChartVersion("rancher-backup", catalog.RancherChartRepo)
+	if err != nil {
+		return err
+	}
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	chartInstallOptions := charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestBackupVersion,
+		ProjectID: project.ID,
+	}
+	chartFeatureOptions := &charts.RancherBackupOpts{
+		VolumeName:                backupRestoreConfig.VolumeName,
+		BucketName:                backupRestoreConfig.S3BucketName,
+		CredentialSecretName:      secretName,
+		CredentialSecretNamespace: backupRestoreConfig.CredentialSecretNamespace,
+		Enabled:                   backupEnabled,
+		Endpoint:                  backupRestoreConfig.S3Endpoint,
+		Folder:                    backupRestoreConfig.S3FolderName,
+		Region:                    backupRestoreConfig.S3Region,
+	}
+
+	_, err = createOpaqueS3Secret(client.Steve)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Installing backup chart")
+	err = charts.InstallRancherBackupChart(client, &chartInstallOptions, chartFeatureOptions, true)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Waiting for backup chart deployments to have expected number of available replicas")
+	err = shepCharts.WatchAndWaitDeployments(client, project.ClusterID, "cattle-resources-system", metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func createOpaqueS3Secret(steveClient *v1.Client) (string, error) {
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Infof("Creating an opaque secret with name: %v", secretName)
+	secretTemplate := secrets.NewSecretTemplate(secretName, backupRestoreConfig.CredentialSecretNamespace, map[string][]byte{"accessKey": []byte(backupRestoreConfig.AccessKey), "secretKey": []byte(backupRestoreConfig.SecretKey)}, corev1.SecretTypeOpaque)
+	createdSecret, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+
+	return createdSecret.Name, err
+}
+
+func createRancherResources(client *rancher.Client, clusterID string, context string) ([]*management.User, []*management.Project, []*management.RoleTemplate, error) {
+	userList := []*management.User{}
+	projList := []*management.Project{}
```

Comment:
is there a v1 version of the project you could switch to instead? I know users are all in v3 but wanted to double check for projects.